src/components/vm/vmRemoveSpiceDialog.jsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React, { useState } from 'react';
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Modal } from "@patternfly/react-core/dist/esm/components/Modal";
+import { Text, TextContent } from "@patternfly/react-core/dist/esm/components/Text";
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { useDialogs } from 'dialogs.jsx';
+
+import { domainRemoveSpice } from '../../libvirtApi/domain.js';
+
+const _ = cockpit.gettext;
+
+export const RemoveSpiceDialog = ({ vmName, vmId, connectionName }) => {
+    const Dialogs = useDialogs();
+    const [error, dialogErrorSet] = useState(null);
+    const [inProgress, setInProgress] = useState(false);
+
+    function onRemove() {
+        setInProgress(true);
+
+        return domainRemoveSpice({ connectionName, id: vmId })
+                .then(() => Dialogs.close())
+                .catch(exc => {
+                    console.warn("Failed to remove spice support:", exc);
+                    dialogErrorSet({
+                        dialogError: _("Failed to remove spice support"),
+                        // FIXME: link to https://access.redhat.com/solutions/6955095
+                        dialogErrorDetail: _("Do this manually FIXME"),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1387-20240112-155815-cbcdc423-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)